test/client/tasks_test.rb

```diff
@@ -0,0 +1,179 @@
+  end
+
+  it "list all in-flight tasks" do
+    h = @tasks.get
+    assert h["nodes"].keys.size > 0
+
+    total_tasks = h["nodes"].map { |k, v| v["tasks"].keys.count }.sum
+    assert total_tasks > 0
+  end
+
+  it "groups by parent->child relationships when get-all tasks API is grouped by 'parents'" do
+    unless $client.version_support.es_version_5_x?
+      skip "Tasks API is not supported in ES version #{$client.version}"
+    end
+
+    h = @tasks.get :group_by => "parents"
+    parent_id = h["tasks"].select { |k, v| v.key?("children") }.keys.first
+    childs_parent_ref = h.dig("tasks", parent_id, "children").first["parent_task_id"]
+    assert_equal parent_id, childs_parent_ref
+  end
+
+  it "raises exception when get_by_id is called without required task & node IDs" do
+    assert_raises(ArgumentError) do
+      @tasks.get_by_id
+    end
+  end
+
+  it "raises exception when get_by_id is called w/invalid task ID is supplied" do
+    node_id = @tasks.get["nodes"].map { |k, v| k }.first
+    assert_raises(ArgumentError) do
+      @tasks.get_by_id node_id, "task_id_should_be_integer"
+    end
+  end
+
+  it "raises exception when get_by_id is called w/invalid node ID is supplied" do
+    assert_raises(ArgumentError) do
+      @tasks.get_by_id nil, 42
+    end
+  end
+
+
+  it "successfully waits for task to complete when wait_for_completion and timeout flags are set" do
+    # do some busy work in bkg thread to keep some tasks running longer
+    Thread.new do
+      begin
+        name = "elastomer-tasks-tests".freeze
+        index = $client.index(name)
+        wait_for_index(name)
+
+        index.docs("person").bulk do |d|
+          (1..100).each do |i|
+            d.index \
+              :foo => "foo",
+              :bar => "bar",
+              :baz => "baz"
+          end
+        end
+        index.refresh
+        Kernel.sleep(0.01)
+      ensure
+        index.delete if index.exists?
+      end
+    end
+
+    # locate task(s) long running enough to be looked up by ID
+    target_tasks = []
+    3.times.each do
+      target_tasks = @tasks.get["nodes"]
+        .map { |k, v| v["tasks"] }
+        .flatten.map { |ts| ts.select { |k, v| /health/ =~ v["action"] } }
+        .flatten.reject { |t| t.empty? }
+      break if target_tasks.size > 0
+    end
+    assert !target_tasks.empty?
+
+    # ensure we can wait on completion of a task
+    success = false
+    target_tasks.each do |t|
+      t = t.values.first
+      resp = @tasks.get_by_id t["node"], t["id"], { :wait_for_completion => true, :timeout => "10s" }
+      success = !resp.key?("node_failures")
+      break if success
+    end
+
+    assert success
+  end
+
+  it "locates the task properly by ID when valid node and task IDs are supplied" do
+    # do some busy work in bkg thread to keep some tasks running longer
+    Thread.new do
+      begin
+        name = "elastomer-tasks-tests".freeze
+        index = $client.index(name)
+        wait_for_index(name)
```

Comment:
Same comment here regarding a new `cilent` for this thread to use. Hrmmm ... a helper method to duplicate a client connection would be super rad. Better yet, implementing the `Elastomer::Client#dup` method would be super helpful. I can do that work in a separate PR.

For now using a different `client` is recommended.